src/Microsoft.Azure.WebJobs.Logging/FunctionInstanceLogItem.cs

```diff
@@ -153,6 +160,57 @@ public void Validate()
                 }
             }
         }
+                     
+        /// <summary>
+        /// Truncate various fields to fit in logging sizes. 
+        /// </summary>
+        public void Truncate()
+        {
+            // This is fundamentally driven by performance. We need to fit log entries into table rows.
+            // Truncate to ensure that we're under table's maximum request payload size (4mb).
+            // None of these limits (except for output log)  should actually get hit in normal scenarios. 
+            this.TriggerReason = Truncate(this.TriggerReason, MaxTriggerReasonLength);
+            this.ErrorDetails = Truncate(this.ErrorDetails, MaxErrorLength);
+
+            // Logger should already have truncated this, but just in case. 
```

Comment:
I added a code-review comment to call that out. It's in the InstanceTableEntity ctor.